packages/sdk/src/transactions/PSBTBuilder.ts

```diff
@@ -0,0 +1,470 @@
+import { networks, Psbt, Transaction } from "bitcoinjs-lib";
+import reverseBuffer from "buffer-reverse";
+
+import { INSTANT_BUY_SELLER_INPUT_INDEX } from "../config";
+import type { Network } from "../config/types";
+import { MINIMUM_AMOUNT_IN_SATS } from "../constants";
+import { FeeEstimator } from "../fee/FeeEstimator";
+import type { InputsToSign } from "../inscription/types";
+import { BaseDatasource } from "../modules/BaseDatasource";
+import { JsonRpcDatasource } from "../modules/JsonRpcDatasource";
+import {
+  convertSatoshisToBTC,
+  generateTxUniqueIdentifier,
+  getNetwork,
+  toXOnly,
+} from "../utils";
+import { InputType, processInput } from "./helper";
+import type { Output, UTXOLimited } from "./types";
+
+export interface PSBTBuilderOptions {
+  address: string;
+  changeAddress?: string;
+  feeRate: number;
+  network: Network;
+  outputs: Output[];
+  publicKey: string;
+  autoAdjustment?: boolean;
+  instantTradeMode?: boolean;
+  datasource?: BaseDatasource;
+}
+
+export type InjectableInput = {
+  injectionIndex: number;
+  txInput: unknown;
+  sats: number;
+  standardInput: InputType;
+};
+
+export interface InjectableOutput {
+  injectionIndex: number;
+  txOutput: unknown;
+  sats: number;
+  standardOutput: unknown;
+}
+
+export class PSBTBuilder extends FeeEstimator {
+  protected address: string;
+
+  protected changeAddress?: string;
+
+  protected changeAmount = 0;
+
+  protected datasource: BaseDatasource;
+
+  protected injectableInputs: InjectableInput[] = [];
+
+  protected injectableOutputs: InjectableOutput[] = [];
+
+  protected inputAmount = 0;
+
+  protected inputs: InputType[] = [];
+
+  protected outputAmount = 0;
+
+  protected outputs: Output[] = [];
+
+  protected psbt: Psbt;
+
+  protected publicKey: string;
+
+  /**
+   * Replace-by-fee (RBF) is a feature that allows users to replace one version of an unconfirmed transaction
+   * with a different version of the transaction that pays a higher transaction fee.
+   * This can be done multiple times while the transaction is unconfirmed.
+   *
+   * Refer to [BIP-125](https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki).
+   */
+  protected rbf = true;
+
+  protected utxos: UTXOLimited[] = [];
+
+  protected usedUTXOs: string[] = [];
+
+  private autoAdjustment: boolean;
+
+  private instantTradeMode: boolean;
+
+  private nativeNetwork: networks.Network;
```

Comment:
@iamcrazycoder is this variable necessary? it's only used once in the constructor and nowhere else

Reply:
Feel free to remove it.